src/sympc/protocol/aby3/aby3.py

```diff
@@ -0,0 +1,161 @@
+        """Initialization of the Protocol.
+
+        Args:
+            security_type : specifies the security level of the Protocol.
+
+        Raises:
+            ValueError : If invalid security_type is provided.
+        """
+        if security_type not in self.security_levels:
+            raise ValueError(f"{security_type} is not a valid security type")
+
+        self.security_type = security_type
+
+    def __eq__(self, other: Any) -> bool:
+        """Check if "self" is equal with another object given a set of attributes to compare.
+
+        Args:
+            other (Any): Object to compare
+
+        Returns:
+            bool: True if equal False if not.
+        """
+        if not self.security_type == other.security_type:
+            return False
+
+        if not type(self) == type(other):
+            return False
+
+        return True
+
+    @staticmethod
+    def trunc1(
+        ptr_list: List["ReplicatedSharedTensor"], shape: torch.Size, session: Session
+    ) -> List["ReplicatedSharedTensor"]:
+        """Performs the ABY3 trunc1 algorithm.
+
+        Args:
+            ptr_list (List[ReplicatedSharedTensor]): Tensors to truncate
+            shape(torch.Size) : shape of tensor values
+            session(Session) : session the tensor belong to
+
+        Returns:
+            List["ReplicatedSharedTensor"] : Truncated shares.
+        """
+        rand_value = torch.empty(size=shape, dtype=session.tensor_type).random_(
+            generator=gen
+        )
+        base = session.config.encoder_base
+        precision = session.config.encoder_precision
+        scale = base ** precision
+        x1 = ptr_list[0].get_copy().shares[0]
+        x2, x3 = ptr_list[1].get_copy().shares
+        x1_trunc = x1 >> precision if base == 2 else x1 // scale
+        x_trunc = (x2 + x3) >> precision if base == 2 else (x2 + x3) // scale
+        shares = [x1_trunc, x_trunc - rand_value, rand_value]
+        ptr_list = ReplicatedSharedTensor.distribute_shares(shares, session)
+        return ptr_list
+
+    @staticmethod
+    def truncate(x: MPCTensor, session: Session) -> MPCTensor:
+        """Perfoms the ABY3 truncation algorithm.
+
+        Args:
+            x (MPCTensor): input tensor
+            session (Session) : session of the input tensor.
+
+        Returns:
+            MPCTensor: truncated MPCTensor.
+
+        Raises:
+            ValueError : parties involved in the computation is not equal to three.
+
+        TODO :Switch to trunc2 algorithm  as it is communication efficient.
+        """
+        if session.nr_parties != 3:
+            raise ValueError("Share trunc1 algorithm works only for 3 parites.")
+
+        share_ptrs = ABY3.trunc1(x.share_ptrs, x.shape, session)
+        result = MPCTensor(shares=share_ptrs, session=session, shape=x.shape)
+
+        return result
+
+    @staticmethod
+    def trunc2(x: MPCTensor, session: Session) -> MPCTensor:
```

Comment:
Q: Could we find a better name for ```trunc1```/```trunc2```?

Reply:
I followed the convention from the ABY3 paper,so it would be easy to refer to, any suggestions for a better name?

Reply:
truncation_algorithm1 and truncation_algorithm2 ?

Reply:
Yep, let's go with this